common/metrics/build.rs

```diff
@@ -1,14 +1,20 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+use std::env;
 use std::process::Command;
 
 /// Save revision info to environment variable
 fn main() {
-    let output = Command::new("git")
-        .args(&["rev-parse", "--short", "HEAD"])
-        .output()
-        .unwrap();
-    let git_rev = String::from_utf8(output.stdout).unwrap();
-    println!("cargo:rustc-env=GIT_REVISION={}", git_rev);
+    match env::var("GIT_REVISION") {
+        Ok(git_rev) => println!("cargo:rustc-env=GIT_REVISION={}", git_rev),
+        Err(_e) => {
+            let output = Command::new("git")
+                .args(&["rev-parse", "--short", "HEAD"])
+                .output()
+                .unwrap();
+            let git_rev = &String::from_utf8(output.stdout).unwrap();
```

Comment:
I think the `&` snuck back in. And can we do something like https://github.com/git/git/blob/master/GIT-VERSION-GEN which checks if the index is dirty and appends "-dirty" to the version?